dataset_loaders/data_augmentation.py

```diff
@@ -580,14 +616,24 @@ def random_transform(x, y=None,
         # Compute amounts
         if crop[0] < h:
             # Do random crop
-            top = np.random.randint(h - crop[0])
+            if crop_mode == 'random':
+                top = np.random.randint(h - crop[0])
+            elif crop_mode == 'smart':
+                if smart_crop_top + crop[0] > h:
```

Comment:
This is not correct. We never want to return an image bigger than the crop_size. I suggest you to change the code that computes `smart_crop_top` and `smart_crop_left` so that it only considers feasible positions. I don't know if computing the fg_mask on the feasible positions only could be enough or if we rather want to find the closest feasible crop to the best one. I'll leave it to you to decide.